src/components/tracked-controls.js

```diff
@@ -35,6 +36,27 @@ module.exports.Component = registerComponent('tracked-controls', {
 
     this.previousControllerPosition = new THREE.Vector3();
     this.updateGamepad();
+    this.onComponentInitialized = bind(this.onComponentInitialized, this);
+  },
+
+  play: function () {
+    this.el.addEventListener('componentinitialized', this.onComponentInitialized);
+  },
+
+  pause: function () {
+    this.el.removeEventListener('componentinitialized', this.onComponentInitialized);
+  },
+
+  onComponentInitialized: function (evt) {
+    if (evt.detail.name !== 'tracked-controls') { return; }
+    var self = this;
+    Object.keys(this.el.components).forEach(function (name) {
+      const component = self.el.components[name];
+      if (component.controllerPresent) {
+        // Emit controllerpresent event.
+        self.el.emit('controllerpresent', {name: name, component: component});
```

Comment:
>  Then I wonder if there's a way for the system to automatically set the tracked-controls component on the entity when the controller becomes available.
You won't know which entity, unless you have a component attached to it.  (that is one job `auto-detect-controllers` was originally doing)